fix: rename react-native-ci-cli to setup-ci after migration (#100)
software-mansion/setup-ci: readme.md

## 🤔 Why?

The problem we noticed is that setting up CI is performed once, needs to be researched every time, and is often trial and error driven. The process is repetitive.

> We created a tool that bootstraps CI with the most used features, which are ready for customization in the future. When you run `npx setup-ci` React Native CI CLI generates GitHub workflows for the most popular CI tasks.

Reviewer: ```suggestion
We created a tool that bootstraps CI with the most used features, which are ready for customization in the future. Running `npx setup-ci` generates GitHub workflows for the most popular CI tasks.
```

- Monorepo structure is supported, but make sure you run the script from the app directory,
  not the monorepo root.

> When using `setup-ci`, you can provide additional flags to modify its default behavior.

Reviewer: So maybe let's replace all setup-ci with npx setup-ci here

Author: done in 44a91c68502ea53e19097a9dfb459cae156d5d6f

## 🔐 Repository secrets

> Some workflows generated by `setup-ci` require setting up repository secrets that can be then accessed by workflows as `${{ secrets.* }}`. Make sure to read logs printed by the script

Reviewer: I wonder if we shouldn't always use npx setup-ci but let's discuss this on daily too

Author: we decided to use `npx setup-ci`
